### DISCUSSION TITLE
Default loading the .env when run astro

### DISCUSSION THREAD

**Discussion by nhattran998** [withastro/roadmap#408] [Proposal]
When building the Astro project, Astro will transform import.meta.env.PRIVATE_ENV to process.env.PRIVATE_ENV. This will work well in CI/CD or Docker env(mask and inject PRIVATE_ENV). But locally, BCS Astro does not load .env like NextJs or Vite, so we cannot serve it in the local environment.

I think we can load .env for dev/serve/build based on the `mode` option as Vite does. It is really straightforward and flexible enough to work with the difference env. 

Hope team can do this in the feature <3.

**Comment by florian-lefebvre** [comment]
I believe this is now supported! If during the build we find the value it is inlined otherwise it's transformed into `process.env`
